src/components/dashboard/sidebar/SubsidiesSummary.jsx

```diff
@@ -85,6 +92,25 @@ const SubsidiesSummary = ({
   const hasAvailableSubsidyOrRequests = (
     hasActiveLicenseOrLicenseRequest || hasAssignedCodesOrCodeRequests || learnerCreditSummaryCardData
   );
+  const [assignmentOnlyLearner, setAssignmentOnlyLearner] = useState(false);
+  useEffect(() => {
+    const data = redeemableLearnerCreditPolicies?.flatMap(item => item?.learnerContentAssignments || []);
+    const assignmentsData = sortAssignmentsByAssignmentStatus(data);
+    const { hasActiveAssignments } = getActiveAssignments(assignmentsData);
+    if (
+      !hasActiveLicenseOrLicenseRequest
+      && !hasAssignedCodesOrCodeRequests
+      && !canEnrollWithEnterpriseOffers
```

Comment:
Note: `canEnrollWithEnterpriseOffers` only says if the customer has > 0 enterprise offers. It does not take into whether those offers are current (non-expired). For example, my test customer has 2 expired offers but `canEnrollWithEnterpriseOffers` is still true. This boolean is used to determine whether to run _any_ offer related logic, but assumes at this point checks for current offers in the appropriate places will be made.

Reply:
@adamstankiewicz Any pointers on how to go about this ... I used canEnrollWithEnterpriseOffers since it is [already being used here](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/4eec3cce09779d58e4929ba6b3958823888b0959/src/components/dashboard/sidebar/SubsidiesSummary.jsx#L131C11-L131C40)

Reply:
X-post from Slack for posterity:

I think you discovered a pre-existing bug, unfortunately. When we expanded the learner portal to take into account expired offers in the UX awhile back, this must have been a miss of a place to update to take into account only current offers.

There is some logic within `useEnterpriseOffers` that filters the list of offers to only current offers. We could expose new properties `currentEnterpriseOffers` and `hasCurrentEnterpriseOffers` from the `UserSubsidyContext` such that your updates to `SubsidySummary` can rely on `hasCurrentEnterpriseOffers` in place of `canEnrollWithEnterpriseOffers`.